tests/UserClientTests.cs

```diff
@@ -502,5 +502,116 @@ private async Task DisableSharedLocationsAsync(ChannelWithConfig channel)
             };
             await _channelClient.PartialUpdateAsync(channel.Type, channel.Id, request);
         }
+
+        [Test]
+        public async Task TestMarkDeliveredAsync()
+        {
+            var markDeliveredOptions = new MarkDeliveredOptions
+            {
+                LatestDeliveredMessages = new List<DeliveredMessageConfirmation>
+                {
+                    new DeliveredMessageConfirmation
+                    {
+                        ChannelCID = "channel1",
+                        MessageID = "message1",
+                    },
+                    new DeliveredMessageConfirmation
+                    {
+                        ChannelCID = "channel2",
+                        MessageID = "message2",
+                    }
+                },
+                UserID = _user1.Id,
+            };
+
+            var response = await _userClient.MarkDeliveredAsync(markDeliveredOptions);
+
+            response.Should().NotBeNull();
+            // Additional assertions can be added based on the expected response structure
+        }
+
+        [Test]
+        public async Task TestMarkDeliveredAsync_WithUserObject()
+        {
+            var markDeliveredOptions = new MarkDeliveredOptions
+            {
+                LatestDeliveredMessages = new List<DeliveredMessageConfirmation>
+                {
+                    new DeliveredMessageConfirmation
+                    {
+                        ChannelCID = "channel1",
+                        MessageID = "message1",
+                    }
+                },
+                User = _user1,
+            };
+
+            var response = await _userClient.MarkDeliveredAsync(markDeliveredOptions);
+
+            response.Should().NotBeNull();
+        }
+
+        [Test]
+        public async Task TestMarkDeliveredAsync_WithUserId()
+        {
+            var markDeliveredOptions = new MarkDeliveredOptions
+            {
+                LatestDeliveredMessages = new List<DeliveredMessageConfirmation>
+                {
+                    new DeliveredMessageConfirmation
+                    {
+                        ChannelCID = "channel1",
+                        MessageID = "message1",
+                    }
+                },
+                UserID = _user1.Id,
+            };
+
+            var response = await _userClient.MarkDeliveredAsync(markDeliveredOptions);
+
+            response.Should().NotBeNull();
+        }
+
+        [Test]
+        public void TestMarkDeliveredAsync_NullData_ThrowsArgumentNullException()
```

Comment:
This test doesn't actually test anything and will always return 'Success'. Please try removing the exceptions from `MarkDeliveredAsync` implementation, and the test will still pass.

The reason is that the async method is not awaited. This test should be written as:
```
        public Task TestMarkDeliveredAsync_NullData_ThrowsArgumentNullException()
        {
            Func<Task> markDeliveredCall = async () => await _userClient.MarkDeliveredAsync(null);

            return markDeliveredCall.Should().ThrowAsync<ArgumentNullException>();
        }
```
This also applies to all tests below that call Async method without the `await`